plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -372,7 +372,9 @@ function perflab_install_and_activate_plugin( string $plugin_slug, array &$proce
 		// Replace new Plugin_Installer_Skin with new Quiet_Upgrader_Skin when output needs to be suppressed.
 		$skin     = new WP_Ajax_Upgrader_Skin( array( 'api' => $plugin_data ) );
 		$upgrader = new Plugin_Upgrader( $skin );
-		$result   = $upgrader->install( $plugin_data['download_link'] );
+		// Remove the version number from the link to download the latest plugin version.
+		$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d+\.\d+\.\d+\.zip#', '$1.zip', $plugin_data['download_link'] );
```

Comment:
It might be safer to do:

```suggestion
		$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d.*.zip#', '$1.zip', $plugin_data['download_link'] );
```

The reason being that it's possible for plugin versions to have `-rc` or `-beta` tagged on. 

For example: 
https://wordpress.org/plugins/rest-api/
=>
https://downloads.wordpress.org/plugin/rest-api.2.0-beta15.zip

Granted, it's highly unlikely that we'll ever release one of our plugins with such a Stable Tag, and it's not super common across the repo: https://wpdirectory.net/search/01JDAFQMKVDE6ME17SH47V33D9

Reply:
So I don't feel strongly about this.

Reply:
Worst case in such a scenario is that the original URL ends up getting used without the version removed, which will likely still work (even if it has the even slighter possibility of downloading a stale version).

Reply:
Thank you for pointing this out! I hadn’t considered the possibility of version tags like `-rc` or `-beta`. That’s an oversight on my part.

However, I believe the regex `'#(\/plugin\/[^\/]+)\.\d.*.zip#'` might not work as intended because it could capture the plugin name greedily. For instance, in the example provided, the download link might end up being `https://downloads.wordpress.org/plugin/rest-api.2.zip` instead of `https://downloads.wordpress.org/plugin/rest-api.zip`.

To address this, one solution could be to modify the regex to capture the plugin name non-greedily, like this:

```php
'#(\/plugin\/[^\/]+?)\.\d.*.zip#'
```

Alternatively, another approach would be to use the following regex:

```php
'#(\/plugin\/[^\/]+)\.\d+\.\d+\.\d+(?:-\w+)?\.zip#'
```

This pattern accounts for optional tags like `-rc` or `-beta` and aligns with the regex used in the `versions` command, so it seems like a good choice to me.

Reply:
Note in the example https://downloads.wordpress.org/plugin/rest-api.2.0-beta15.zip there is no patch version either, just a major and minor. So that would need to be optional as well.

This should address the greediness issue, by using `\.\d.*?.zip` instead of `\.\d.*.zip`:

```php
$download_link = (string) preg_replace( '#(\/plugin\/[^\/]+)\.\d.*?.zip#', '$1.zip', $plugin_data['download_link'] );
```